File: app/src/main/java/com/exercise/davismiyashiro/popularmovies/data/Response.java
package com.exercise.davismiyashiro.popularmovies.data;

import java.util.List;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by Davis Miyashiro on 05/02/2017.
 */

public class Response<T> {

    @SerializedName("page")
    @Expose
    private Integer page;
    @SerializedName("results")
    @Expose
    private List<T> results = null;
    @SerializedName("total_results")
    @Expose
    private Integer totalResults;
    @SerializedName("total_pages")
    @Expose
    private Integer totalPages;

    public Integer getPage() {
        return page;
    }

    public void setPage(Integer page) {
        this.page = page;
    }

    public List<T> getResults() {
        return results;
    }

    public void setResults(List<T> results) {
        this.results = results;
    }

    public Integer getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(Integer totalResults) {
        this.totalResults = totalResults;
    }

    public Integer getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(Integer totalPages) {
        this.totalPages = totalPages;
    }

}

File: app/src/main/java/com/exercise/davismiyashiro/popularmovies/data/Response.kt
package com.exercise.davismiyashiro.popularmovies.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Created by Davis Miyashiro on 05/02/2017.
 */

@JsonClass(generateAdapter = true)
data class Response<T>(
        @field:Json(name = "page") var page: Int = 0,
        @field:Json(name = "results") var results: List<T>,
        @field:Json(name = "total_results") var totalResults: Int = 0,
        @field:Json(name = "total_pages") var totalPages: Int = 0
)